### ISSUE TITLE
Is this somehow related to keras-contrib?

### ISSUE THREAD

**Issue by PhilipMay** [titu1994/BatchRenormalization#5]
Hi,

is this work here somehow related to keras-contrib? See here:
https://github.com/keras-team/keras-contrib

There is also a Batch Renorm implementation (which has known bugs): https://github.com/keras-team/keras-contrib/blob/master/keras_contrib/layers/normalization.py#L155

Thanks
Philip

**Comment by titu1994** [comment]
Yes, this repository was the base for the Contrib version. As of a few months ago, the Contrib version supercedes this local version. 

Though I have not heard of the bug with Tensorflow backend. Could you possibly elaborate?

**Comment by PhilipMay** [comment]
Hi @titu1994 
thanks for the fast answer. For the bug see here: 
https://github.com/keras-team/keras-contrib/blob/master/keras_contrib/layers/normalization.py#L224

**Comment by titu1994** [comment]
Yes I read that. I was wondering what the actual bug was. Is it an issue in shapes, or wrong updates or something. 

I haven't looked into batch renormalization since I implemented Group Normalization.

**Comment by PhilipMay** [comment]
> I haven't looked into batch renormalization since I implemented Group Normalization.

You talk about this here right? https://github.com/titu1994/Keras-Group-Normalization

**Comment by titu1994** [comment]
Yes.

**Comment by PhilipMay** [comment]
> Yes.

Ahh ok nice. Thanks for the reply.
Did you do some experiments with Group N.? How is it compared to BN and Batch Renormalization?

**Comment by titu1994** [comment]
It's good for batch sizes less than 8 in the few experiments I did. Anything above that and batch norm beats it at test time. 

But not by a huge amount or anything. 0.5-1% difference between the two. 

Haven't compared against batch renorm, but according to the group norm paper, batch renorm behaves close to batch norm performance for most batch sizes.

**Comment by PhilipMay** [comment]
Ok. Thanks! :-)

**Comment by PhilipMay** [comment]
Closing this...
